src/engine/fileOpsIns.cpp

```diff
@@ -810,16 +1008,148 @@ void DivEngine::loadFF(SafeReader& reader, std::vector<DivInstrument*>& ret, Str
 }
 
 void DivEngine::loadOPM(SafeReader& reader, std::vector<DivInstrument*>& ret, String& stripPath) {
-  DivInstrument* ins[128];
-  memset(ins,0,128*sizeof(void*));
+  std::vector<DivInstrument*> insList;
+
+  int readCount = 0;
+
+  bool patchNameRead = false,
+       lfoRead = false,
+       characteristicRead = false,
+       m1Read = false,
+       c1Read = false,
+       m2Read = false,
+       c2Read = false;
+
+  DivInstrument* newPatch = nullptr;
+  
+  auto completePatchRead = [&]() {
+    return patchNameRead && lfoRead && characteristicRead && m1Read && c1Read && m2Read && c2Read;
+  };
+  auto resetPatchRead = [&]() {
+    patchNameRead = lfoRead = characteristicRead = m1Read = c1Read = m2Read = c2Read = false;
+    newPatch = nullptr;
+  };
+  auto readIntStrWithinRange = [](String&& input, int limitLow, int limitHigh) {
+    int x = atoi(input.c_str());
+    return (x>limitHigh) ? limitHigh :
+           (x<limitLow) ? limitLow : x;
+  };
+  auto readOpmOperator = [&](SafeReader& reader, DivInstrumentFM::Operator& op) {
+    op.ar = readIntStrWithinRange(reader.readString_Token(), 0, 31);
+    op.dr = readIntStrWithinRange(reader.readString_Token(), 0, 31);
+    op.d2r = readIntStrWithinRange(reader.readString_Token(), 0, 31);
+    op.rr = readIntStrWithinRange(reader.readString_Token(), 0, 31);
+    op.sl = readIntStrWithinRange(reader.readString_Token(), 0, 15);
+    op.tl = readIntStrWithinRange(reader.readString_Token(), 0, 127);
+    op.rs = readIntStrWithinRange(reader.readString_Token(), 0, 3);;
+    op.mult = readIntStrWithinRange(reader.readString_Token(), 0, 15);
+    op.dt = readIntStrWithinRange(reader.readString_Token(), 0, 7);
+    op.dt = (op.dt >= 4) ? (7 - op.dt) : (op.dt + 3);
+    op.dt2 = readIntStrWithinRange(reader.readString_Token(), 0, 3);
+    op.am = readIntStrWithinRange(reader.readString_Token(), 0, 1);
+  };
 
   try {
-    String line;
-        
+    reader.seek(0, SEEK_SET);
+    while (!reader.isEOF()) {
+      String token = reader.readString_Token();
+      if (token.length() == 0) {
+        continue;
+      }
+
+      if (token.compare(0,2,"//") == 0) {
+        if (!reader.isEOF()) {
+          reader.readString_Line();
+        }
+        continue;
+      }
+
+      // At this point we know any other line would be associated with patch params
+      if (newPatch == nullptr) {
```

Comment:
NULL

Reply:
That's a C anachronism, but changed as requested. Hopefully future compiler standards won't turn this into a `-Werror` :P

Some reading:
- C23 proposal to move away from `NULL`: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2312r0.pdf
- Numerous SO threads on the topic and well-documented issues of `NULL` vs `nullptr` re: integer vs pointer overloads: https://stackoverflow.com/questions/20509734/null-vs-nullptr-why-was-it-replaced

Reply:
The main issue with `nullptr` is that it's 3 characters longer.
I take 0.7 seconds to type `NULL`, but 1.0 seconds to type `nullptr`.

Reply:
If I had to nitpick, I'd probably go with C++ standard conventions than individual convenience, as IDEs would generally reduce the TTTT (Time Taken To Type) metric described to be equivalent.